### PR TITLE
Bugfix mask cgi escaped tokens

### PR DESCRIPTION
**Pull request by mkaydev**
Added additional calls to logger to mask cgi escaped (percent encoded) api tokens from logs.

### CODE REVIEW

**Comment by apurvis** [comment]
Also can you add something to the spec?

**Comment by apurvis** [comment]
Also also the build failed

**Comment by mkaydev** [comment]
I added the OR condition to the regex instead of repeating the blocks and added a spec file for the logger with tests for masking the tokens.

I ran the spec task before submitting the pull request, no test fails, 1 is pending.

The failing build[1] shows:

`"NoMethodError: undefined method `last_comment' for #<Rake::Application:0x0000000233cc30>"`

Running the specs shows a deprecation warning:
```
[DEPRECATION] `last_comment` is deprecated.  Please use `last_description` instead.
~/.rvm/rubies/ruby-2.3.3/bin/ruby -S rspec spec/configuration_spec.rb spec/logger_spec.rb spec/resource_spec.rb
```

last_comment doesn't appear in the code. A quick search[2] indicates that rspec-rails would have to be updated.

I can do that if you don't have any objections.

[1] https://travis-ci.org/jarthod/survey-gizmo-ruby/jobs/182420818
[2] http://stackoverflow.com/questions/36733070/last-comment-is-deprecated-please-use-last-description-instead

File: lib/survey_gizmo/logger.rb [hunk 1]
```diff
@@ -2,11 +2,21 @@
...
   class Logger < ::Logger
-    def format_message(severity, timestamp, progname, msg)
-      msg.gsub!(/#{Regexp.quote(SurveyGizmo.configuration.api_token)}/, '<SG_API_KEY>') if SurveyGizmo.configuration.api_token
-      msg.gsub!(/#{Regexp.quote(SurveyGizmo.configuration.api_token_secret)}/, '<SG_API_SECRET>') if SurveyGizmo.configuration.api_token_secret
+    def format_message(severity, timestamp, progname, message)
+      api_token = SurveyGizmo.configuration.api_token
+      api_token_secret = SurveyGizmo.configuration.api_token_secret
 
-      "#{timestamp.strftime('%Y-%m-%d %H:%M:%S')} #{severity} #{msg}\n"
+      message.gsub!(
```
Comment (apurvis) [inline lib/survey_gizmo/logger.rb]: if you want to multiline the gsub for clarity, let's style it like this:

```ruby

if (api_token = SurveyGizmo.configuration.api_token)
  message.gsub!(
    /#{Regexp.quote(api_token)}|#{Regexp.quote(CGI.escape(api_token))}/,
    '<SG_API_KEY>'
  )
end

if (api_token_secret = SurveyGizmo.configuration.api_token_secret)
  message.gsub!(
    /#{Regexp.quote(api_token_secret)}|#{Regexp.quote(CGI.escape(api_token_secret))}/,
    '<SG_API_SECRET>'
  )
end
```

**Comment by apurvis** [comment]
also can you bump the bugfix version number and put a comment in the changelog?

**Comment by apurvis** [comment]
re: the build - are you sure it's `rspec-rails` and not the new major release of `rake` itself?  in travis i see `Installing rake 12.0.0` but in my local `Gemfile.lock` I have `rake (10.4.2)` - that's 2 version bumps.

just trying to be helpful though; i don't care how you get travis to build it is as long as it builds.

**Comment by mkaydev** [comment]
It seems to have been the combination.

rake >=11.0.1 removed the last_comment method and rspec 2.11.0 still used that method.

I updated the rake dependency to version 12 (to reproduce the issue) and then rspec to version 3.4 to fix the issue.

I had to change some of the specs, because the rspec should syntax has been deprecated.

**Comment by apurvis** [comment]
amazing; thanks.  i'll merge this and @jarthod  can push a release

**Comment by apurvis** [comment]
p.s. i did `squash and merge` so @mkaydev you should delete your branch
p.p.s. if you had the willpower it would be awesome to fix all the build warnings.  not a problem really if not just throwing that out there

**Comment by jarthod** [comment]
Thanks @mkaydev and @apurvis, I just released `6.2.11`

**Pull request merged**